fedimintd/src/fedimintd.rs

```diff
@@ -265,7 +265,7 @@ impl Fedimintd {
             code_version_hash: code_version_hash.to_owned(),
             code_version_str: code_version_vendor_suffix.map_or_else(
                 || fedimint_version.to_string(),
-                |suffix| format!("{fedimint_version}.{suffix}"),
+                |suffix| format!("{fedimint_version}-{suffix}"),
```

Comment:
If we want this to pass semver, then we should probably use a `+` which means "build" which matches our meaning better than `-` "prerelease".

Reply:
Good point, otherwise it might be colliding with RCs